docs/architecture/decisions/00xx-python-gardenlinux-lib.md

```diff
@@ -0,0 +1,38 @@
+# xx. Organizing Python Code for Garden Linux
+
+Date: 2025-11-28
+
+## Status
+
+Proposed
+
+## Context
+
+Historically, Python code for Garden Linux resided in the main repository, often without proper testing or organization. Some scripts were embedded directly in GitHub Actions workflows, while others were placed in the `hack/` directory. Dependency management for this code was inconsistent and lacked a systematic approach.
+
+Recently, a separate repository, `python-gardenlinux-lib`, was introduced to host Python code for various usages. The work done to implement this separation brought benefits such as dedicated tests and linters, but these improvements are not inherently tied to having a separate repository.
+
+Currently, the lib repo contains code used in GitHub Actions workflows and feature-related logic. This introduces an indirection, making debugging and modifying pipeline code more complex, as changes require updates in the lib repo, releasing a new version, and then updating the dependency in the main repo.
+
+Examples for scope creep in the lib:
+
+- [exportLibs.py](https://github.com/gardenlinux/python-gardenlinux-lib/pull/253)
```

Comment:
Main reason for the `exportLibs.py` PR have been to benefit from the `python-gardenlinux-lib` repository structure and tooling. It allowed to reuse third-party packages instead of reinventing the wheel for a specific use case.

I see this example as a first step towards supporting tools to build and mantain the GardenLinux repository and code as an `gl-build` tool (for example) that allows advanced logic to be implemented in a structured way with code quality and version maintenance in mind.

Of course this and other tools could be splitted from pure library code and live in it's own repository reusing existing code.